### DISCUSSION TITLE
NestJS code-first gql in plugins

### DISCUSSION THREAD

**Discussion by jbeckton** [vendurehq/vendure#1899] [General]
Is it possible to use NestJS code-first approach for graphQL within plugin implementations?

**Comment by michaelbromley** [comment]
Hi Jesse,
This is not currently possible. There has been a bit of discussion in this thread: https://github.com/vendure-ecommerce/vendure/discussions/1326

I'm certainly open to supporting this, but I'm not clear on what it will take to get there. As I state in the linked thread, I'm happy is someone wants to explore a proof-of-concept for this. For me it's not a priority to put time into this at the moment. After v2 ships I would consider revisiting.

**Reply by jbeckton** [reply]
Understood, lets get v2 out the door. I'm just trying to avoid writing gql schemas manually. I like NestJS so I'll take a peek at what needs to be done to make this work with Vendure. Thanks!

**Comment by jbeckton** [comment]
@michaelbromley Looks like if you converted vendure core to use code-first approach you could use autoSchema to generate the whole schema (plugins too) at run time into memory.

````
GraphQLModule.forRoot({
  include: [...allModulesWithGql],
 autoSchemaFile: true
}),
````

**Reply by michaelbromley** [reply]
OK thanks for alerting me to this. I'll revisit after I get all the dependencies updated as part of v2 in the new year.

**Reply by jbeckton** [reply]
Looks like my links to the NestJS docs got stripped from the post, my apologies. If I need to I can dig them up again.
